golang/cosmos/vm/action.go

```diff
@@ -0,0 +1,81 @@
+package vm
+
+import (
+	"reflect"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Jsonable is a value, j, that can be passed through json.Marshal(j).
+type Jsonable interface{}
+
+// ActionPusher enqueues data for later consumption by the controller.
+type ActionPusher func(ctx sdk.Context, action Jsonable) error
+
+// WithActionContext interprets `default:"..."` tags and adds other fields from
+// the SDK context to the action.
+func WithActionContext(ctx sdk.Context, action Jsonable) Jsonable {
```

Comment:
Consider renaming to `PopulateAction()`. The "with...context" triggers me to think of the various "with" methods for `sdk.Context`.